pkg/clusteragent/admission/mutate/config/config.go

```diff
@@ -241,6 +241,10 @@ func injectExternalDataEnvVar(pod *corev1.Pod) (injected bool) {
 			_ = log.Errorf("Cannot inject identity into nil container")
 			continue
 		}
+		if common.Contains(containerInjection.container.Env, ddExternalDataEnvVarName) {
```

Comment:
why not using `InjectEnv()` function instead?

Reply:
Checked together, we improved implementation by adding a `common.InjectDynamicEnv` function.